sklearn/manifold/t_sne.py

```diff
@@ -680,6 +684,11 @@ def _fit(self, X, skip_num_points=0):
                 raise ValueError("All distances should be positive, the "
                                  "precomputed distances given as X is not "
                                  "correct")
+        if not ((self.metric_params is None) or
+                ((type(self.metric_params) == dict) and
+                (len(self.metric_params) > 0))):
+            raise TypeError("'metric_params' parameter must be either None or\
```

Comment:
This edge case should be tested in `sklearn/manifold/tests/test_t_sne.py`.

Reply:
Please avoid line continuation with \. The right way to do this in python is like this:
```
Blah("here are some words that "
     "form a string.")
```

Reply:
I think it is unfriendly to reject an empty dict. It may also be unfriendly to reject subtypes of dict or other implementations of the Mapping interface. Much better off duck typing than over-validating.